zanshinsdk/test_client.py

```diff
@@ -2068,3 +2081,120 @@ def test_validate_uuid_invalid(self):
+        boto3_profile='foo'
+
+        try:
+            self.sdk.onboard_scan_target(boto3_profile, region, organization_id, kind, name, credential, schedule)
+        except Exception as e:
+            self.assertEqual(str(e), "Onboard doesn't support AZURE environment yet")
+    
+    @unittest.skipIf('HAVE_BOTO3', "requires not have boto3")
+    def test_onboard_scan_target_aws_missing_boto3(self):
+        organization_id = "822f4225-43e9-4922-b6b8-8b0620bdb1e3"
+        kind = zanshinsdk.ScanTargetKind.AWS
+        name = "OnboardTesting-it"
+        credential = zanshinsdk.ScanTargetAWS("4321")
+        schedule = "0 0 * * *"
+        region = "us-east-1"
+        boto3_profile='foo'
+
+        try:
+            self.sdk.onboard_scan_target(boto3_profile, region, organization_id, kind, name, credential, schedule)
+        except Exception as e:
+            self.assertEqual(str(e), "boto3 not present. boto3 is required to perform AWS onboard.")
+
+    @unittest.skipUnless('HAVE_BOTO3', "requires boto3")
+    @mock_sts
+    def test_onboard_scan_target_aws_invalid_credentials(self):
+        self.mock_aws_credentials()
+        organization_id = "822f4225-43e9-4922-b6b8-8b0620bdb1e3"
+        kind = zanshinsdk.ScanTargetKind.AWS
+        name = "OnboardTesting-it"
+        credential = zanshinsdk.ScanTargetAWS("4321")
+        schedule = "0 0 * * *"
+        region = "us-east-1"
+        boto3_profile='non_default'
+
+        try:
+            self.sdk.onboard_scan_target(boto3_profile, region, organization_id, kind, name, credential, schedule)
+        except Exception as e:
+            self.assertEqual(str(e), "boto3 session is invalid. Working boto3 session is required.")
+
+    
+    @unittest.skipUnless('HAVE_BOTO3', "requires boto3")
+    @patch("zanshinsdk.client.isfile")
+    @patch("zanshinsdk.Client._request")
+    @mock_sts
+    @mock_cloudformation
+    @mock_s3
+    def test_onboard_scan_target_aws(self, request, mock_is_file):
+        import boto3, json
+        
+        
+        # Setup test data
+        aws_account_id = "123456789012"
+        organization_id = "822f4225-43e9-4922-b6b8-8b0620bdb1e3"
+        created_scan_target_id = "14f79567-6b68-4e3a-b2f2-4f1383546251"
+        kind = zanshinsdk.ScanTargetKind.AWS
+        name = "OnboardTesting-it"
+        credential = zanshinsdk.ScanTargetAWS(aws_account_id)
+        schedule = "0 0 * * *"
+        region = "us-east-1"
+        boto3_profile = 'foo'
+
+        # Mock AWS Credentials for Boto3
+        self.mock_aws_credentials()
+        
+        # Mock request to create new Scan Target
+        mock_is_file.return_value = True
+        _data = f"[default]\napi_key=api_key"
+
+        with patch("__main__.__builtins__.open", mock_open(read_data=_data)):
+            request.return_value = Mock(status_code=200, json=lambda: {
+                                        "id": created_scan_target_id})
+            self.sdk = zanshinsdk.Client()
```

Comment:
It is preferable to use a local variable instead of overwriting the instance created in setUp

Reply:
Fixed! Thanks.